tests/python/pants_test/backend/python/test_python_chroot.py

```diff
@@ -82,6 +82,38 @@ def dumped_chroot(self, targets):
       finally:
         python_chroot.delete()
 
+  def test_shebang_modified(self):
+    self.create_file(relpath='src/python/test/main.py', contents=dedent("""
+    def main():
+      print('Hello World!')
+    """))
+
+    binary = self.make_target(spec='src/python/test',
+                              target_type=PythonBinary,
+                              source='main.py',
+                              shebang='/usr/bin/env python2'
+                             )
+
+    with self.dumped_chroot([binary]) as (pex_builder, python_chroot):
+      pex_builder.freeze()
+      self.assertEqual(python_chroot._targets[0].shebang, '/usr/bin/env python2')
```

Comment:
The chroot dumping here is superfluous. This test could actually just be:
```python
binary = self.make_target(spec='src/python/test',
                          target_type=PythonBinary,
                          shebang='/usr/bin/env python2')
self.assertEqual(binary.shebang, '/usr/bin/env python2')
```

... and that's not really worth a test (nor is the test in the right spot - should probably be in `tests/python/pants_test/backend/python/targets/test_python_binary.py`).

How about adding to https://github.com/pantsbuild/pants/blob/master/tests/python/pants_test/backend/python/tasks2/test_python_binary_create.py

You could then build a pex and maybe read it as a flat file, asserting the first line of that file is in fact the expected shebang.

Reply:
Removed the two old tests and added a new one as suggested. 
The state of the tests inside the v2 pipeline is actually pretty nice and readable. 👍